packages/engine-render/src/components/sheets/sheet-skeleton.ts

```diff
@@ -502,10 +527,13 @@ export class SpreadsheetSkeleton extends Skeleton {
         return this;
     }
 
-    calculate(bounds?: IViewportInfo): Nullable<SpreadsheetSkeleton> {
+    /**
+     * Refresh cache after markDirty by SheetSkeletonManagerService.reCalculate()
+     * @param bounds
+     */
+    calculate(): Nullable<SpreadsheetSkeleton> {
         this._resetCache();
-
-        this.setStylesCache(bounds);
+        this._updateLayout();
```

Comment:
does this one maybe cause performance problem?  It seems the sekelton dirty is not equal to layout dirty.  Do we have some state flag to mark the layout change?

Reply:
known performance issue